openwisp_controller/config/base/device.py

```diff
@@ -279,6 +285,9 @@ def save(self, *args, **kwargs):
                 self.key = self.generate_key(shared_secret)
         state_adding = self._state.adding
         super().save(*args, **kwargs)
+        # Triggering the whois lookup for new devices and old devices based
+        # on last_ip is changed or not
+        self.trigger_whois_lookup()
```

Comment:
This comment would be better placed in the function’s docstring. Additionally, we can make it configurable and trigger it only when the feature is enabled.